src/test/java/org/sagebionetworks/bridge/services/FileServiceTest.java

```diff
@@ -218,5 +257,126 @@ public void deleteAllStudyFiles() {
+        assertEquals(2, captured.getItems().size());
+        assertEquals(DOWNLOAD_URL_1, captured.getItems().get(0).getDownloadURL());
+        assertEquals(DOWNLOAD_URL_2, captured.getItems().get(1).getDownloadURL());
+        
+        verify(mockFileRevisionDao).getFileRevisions(GUID, 100, 25);
+    }
+    
+    @Test(expectedExceptions = BadRequestException.class)
+    public void getFileRevisionsPageSizeTooSmall() {
+        FileMetadata metadata = new FileMetadata();
+        doReturn(metadata).when(service).getFile(TEST_STUDY, GUID);
+        
+        service.getFileRevisions(TEST_STUDY,GUID, 0, 1);
+    }
+    
+    @Test(expectedExceptions = BadRequestException.class)
+    public void getFileRevisionsSizeTooLarge() {
+        FileMetadata metadata = new FileMetadata();
+        doReturn(metadata).when(service).getFile(TEST_STUDY, GUID);
+        
+        service.getFileRevisions(TEST_STUDY, GUID, 0, 1000);
+    }
+    
+    @Test
+    public void createFileRevision() throws Exception {
+        FileMetadata metadata = new FileMetadata();
+        doReturn(metadata).when(service).getFile(TEST_STUDY, GUID);
+        
+        URL url = new URL("https://" + UPLOAD_BUCKET);
+        when(mockS3Client.generatePresignedUrl(any())).thenReturn(url);
+        
+        FileRevision revision = new FileRevision();
+        revision.setName("name.pdf");
+        revision.setFileGuid(GUID);
+        revision.setMimeType("application/pdf");
+        
+        FileRevision returned = service.createFileRevision(TEST_STUDY, revision);
+        assertEquals(returned.getCreatedOn(), TIMESTAMP);
+        assertEquals(returned.getStatus(), PENDING);
+        assertEquals(returned.getUploadURL(), "https://" + UPLOAD_BUCKET);
+        assertEquals(returned.getDownloadURL(), "https://docs.sagebridge.org/oneGuid.1422319112486");
+        
+        verify(mockS3Client).generatePresignedUrl(requestCaptor.capture());
+        GeneratePresignedUrlRequest request = requestCaptor.getValue();
+        assertEquals(request.getExpiration().getTime(), TIMESTAMP.plusMinutes(EXPIRATION_IN_MINUTES).getMillis());
+        assertEquals(request.getBucketName(), UPLOAD_BUCKET);
+        assertEquals(request.getKey(), GUID + "." + TIMESTAMP.getMillis());
+        assertEquals(request.getContentType(), "application/pdf");
+        assertEquals(request.getResponseHeaders().getContentDisposition(), "attachment; filename=\"name.pdf\"");
+        
+        verify(mockFileRevisionDao).createFileRevision(revisionCaptor.capture());
+        assertSame(revisionCaptor.getValue(), returned);
+    }
+    
+    @Test(expectedExceptions = EntityNotFoundException.class, 
+            expectedExceptionsMessageRegExp=".*FileMetadata not found.*")
+    public void createFileRevisionNotFound() throws Exception {
+        FileRevision revision = new FileRevision();
+        revision.setFileGuid(GUID);
+        
+        service.createFileRevision(TEST_STUDY, revision);
+    }
```

Comment:
Added this, moved around the checks so that we don't trigger an NPE before discovering the revision has no file GUID (as called from the controller, there will always be something in the fileGuid field though).

Reply:
(There's no update call... finish changes two fields on an existing record with no user input.)